### ISSUE TITLE
[macOS] Crash in multi-window mode in the Controls.sample project 

### ISSUE THREAD

**Issue by MartyIX** [dotnet/maui#21306]
### Description

Setting window size crashes the MAUI application.

### Steps to Reproduce

Compile and run `Controls.Sample` project:

```bash
cd src/Controls/samples/Controls.Sample
dotnet build -t:run --configuration Debug -p:RuntimeIdentifier=maccatalyst-arm64 --framework net8.0-maccatalyst
```

In the app, navigate to Core -> Multi-Window and click `Set window size to 700x500` and the application crashes.

### Version with bug

Nightly / CI build (93940d6f931a54bce074f8dfd2013440e0827cf3). 

However, I can see something similar in my MAUI app as well but reproducing the issue in the Controls.Sample project is easiest so I reported it this way.

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

macOS

### Affected platform versions

Apple Mac M1 Sonoma 14.4

### Did you find any workaround?

No.

**Comment by drasticactions** [comment]
It's recursion.

https://github.com/dotnet/maui/blame/93940d6f931a54bce074f8dfd2013440e0827cf3/src/Controls/src/Core/Window/Window.cs#L230-L234

https://github.com/dotnet/maui/pull/16637

This PR introduced setting the WidthProperty and HeightProperty in this method. That then triggers a FrameChange, which then triggers the width/height to update.

https://github.com/dotnet/maui/blob/5fa7b3d4f44b773399bf4b3dd2fe2266199e10fb/src/Core/src/Platform/iOS/WindowExtensions.cs#L33-L34

So it keeps going until a stack overflow. So there probably needs to be a code change there I think? Note though that even if fixed, it still probably won't work, https://github.com/dotnet/maui/issues/9704. Mac Catalyst doesn't expose direct APIs for controlling Window size.

CC @mattleibow

**Comment by mattleibow** [comment]
I just tried the new code in  #22413 and it is not fixed - we will probably need this fix somehow.

I wonder if there is a general fix we can do when during a value set in xaml, the handler "rejects" it. For example, if there is some OS restriction on a window size or some property state. The handler may want to say no and reset the value.

Maybe the current implementation of this is wrong and the handler must just ignore invalid things and then somehow trigger some "next UI loop" reset - or we need a special flag in xplat code like the PR #21357 was doing. However, if we start adding flags then we will have flags all over. This could happen if you were to say set some property on a button (eg enabled) but the OS disagrees because it is some protected button - then the handler will have to trigger the old value. This is a bad example, but hopefully shows that right now we got frame, but it could be any property.

But, how likely is all this? Maybe frame is the only one so the flag is fine. I think the issue will happen with all but Windows as neither Android nor iOS support this. So the flag probably needs to not have any conditions and sime handlers may update frame during the set, we probably need to do the FrameChanged implementation slightly different.
